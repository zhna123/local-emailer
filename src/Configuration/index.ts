import {googleToken} from "./GoogleToken";
import {recipientConfiguration} from "./Recipients";

export interface Configuration {
  defaultRecipient: string;
  recipients: Recipient[];
  googleAuth: any;
  sender: string;
}

export interface Recipient {
  email: string;
  name: string;
  selected: boolean;
  id: string
}

export const configuration: Configuration = {
  ...recipientConfiguration,
  googleAuth: {
    gmailEndpoint: `https://www.googleapis.com/upload/gmail/v1/users/${recipientConfiguration.sender}/messages/send`,
    tokenEndpoint: "https://www.googleapis.com/oauth2/v4/token",
    tokenRequestBody: googleToken,
  },
};