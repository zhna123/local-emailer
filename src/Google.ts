
import { Buffer } from "buffer";
import qs from "qs";
import { configuration } from "./Configuration";

async function getAccessToken() {
  const parameters: RequestInit = {
    method: "POST",
    headers: {
      "Content-Type": "application/x-www-form-urlencoded",
    },
    body: qs.stringify({
      ...configuration.googleAuth.tokenRequestBody,
    }),
  };
  const fetchResult = await fetch(configuration.googleAuth.tokenEndpoint, parameters);
  const res = await fetchResult.json();
  return res.access_token;
}

export function formatEmail(sender: string, recipient: string, subject: string, body: string) {
  return `To: ${recipient}
From: ${sender}
Subject: ${handleUnicode(subject)}

${body ? body : "."}\n`;
}

// https://github.com/googleapis/google-api-nodejs-client/issues/739
// tslint:disable-next-line:max-line-length
// https://github.com/googleapis/google-api-nodejs-client/pull/1088/files/7dd0d7770911688a5e6018d9418381be81253146#diff-ed4183b64bd56b93f40d1aaa8a7597d0
function handleUnicode(stringValue) {
  const buffer = Buffer.from(stringValue);
  const encoded = buffer.toString("base64");
  const full = `=?utf-8?B?${encoded}?=`;
  return full;
}

export async function sendEmail(sender: string, recipient: string, subject: string, body: string) {
  const bearerToken = await getAccessToken();
  const authorizationHeader = `Bearer ${bearerToken}`;
  const parameters: RequestInit = {
    method: "POST",
    headers: {
      "content-type": "message/rfc822",
      authorization: authorizationHeader,
    },
    body: formatEmail(sender, recipient, subject, body),
  };
  // console.log(`Sending email with parameters: ${JSON.stringify(parameters, null, 2)}`);
  const fetchResult = await fetch(configuration.googleAuth.gmailEndpoint, parameters);
  // console.log(`Send email result: ${JSON.stringify(fetchResult, null, 2)}`);
  const fetchResultJson = await fetchResult.json();
  // console.log(`Send email result JSON: ${JSON.stringify(fetchResultJson, null, 2)}`);
}