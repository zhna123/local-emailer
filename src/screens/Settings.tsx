import { View, Text } from 'react-native';
import { rgbStrings as bases } from "solarizer";
import {Picker} from '@react-native-picker/picker';
import { useEffect, useState } from 'react';
import { recipientConfiguration } from '../Configuration/Recipients';
import { Recipient } from '../Configuration';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { RootDrawerScreenProps } from '../../types';


export default function Settings({ route, navigation }: RootDrawerScreenProps<'Settings'>) {
  
  const [selectedRecipient, setSelectedRecipient] = useState('');
  const [recipients, setRecipients] = useState<Recipient[]>([])

  useEffect(() => {
    setRecipients(recipientConfiguration.recipients)
  }, [])

  useEffect(() => {
    const getData = async () => {
      try {
        const value = await AsyncStorage.getItem('default-recipient');
        if (value !== null) {
          setSelectedRecipient(value)
        }
      } catch (e) {
        console.log(`settings-error reading async storage value: ${e}`)
      }
    };
    getData()
  }, [])

  // store the value each time selectedRecipient changes
  useEffect(() => {
    const storeData = async (value: string) => {
      try {
        if (value !== '') {
          await AsyncStorage.setItem('default-recipient', value);
        }
      } catch (e) {
        console.log(`error storing default recipient: ${value}, error: ${e}`)
      }
    };
    storeData(selectedRecipient)
  }, [selectedRecipient])

  const showPickerItems = recipients.map(recipient => {
    return <Picker.Item key={recipient.id} label={recipient.name} value={recipient.name} 
      style={{backgroundColor: bases.base00, color: "white"}} />
  })
  
  return (
    <View style={{ flex: 1, backgroundColor: bases.base03, padding: 20 }}>
      <Text style={{color: "white", fontSize: 18, backgroundColor: bases.base01, padding: 10}}>Default Recipient</Text>

      <Picker
        selectedValue={selectedRecipient}
        onValueChange={(itemValue, itemIndex) =>
          setSelectedRecipient(itemValue)
        }  
        style={{color: "white", fontSize: 16, backgroundColor: bases.base00}}
        dropdownIconColor={"white"}
        mode='dropdown'
      >
        {showPickerItems}
      </Picker>
      
    </View>
  )
}