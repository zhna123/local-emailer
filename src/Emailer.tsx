import React, { useEffect, useState } from "react";
import {
  useWindowDimensions,
  StyleSheet,
  TextInput,
  ToastAndroid,
  View,
  ScrollView,
  Platform
} from "react-native";

import AwesomeButton from "react-native-really-awesome-button";

import { createBasesFromColor, rgb, rgbStrings as bases } from "solarizer";
import { configuration, Recipient } from "./Configuration";
import { sendEmail as send } from "./Google";

const blue = createBasesFromColor(rgb.blue, "base01");
const red = createBasesFromColor(rgb.red, "base01");
const green = createBasesFromColor(rgb.green, "base01");

interface RecipientDict {
  [index: string]: Recipient
}

export const Emailer: React.FC = () => {
  
  const [recipients, setRecipients] = useState<RecipientDict>({})
  const [subject, setSubject] = useState<string>('')
  const [body, setBody] = useState<string>('')
  const [sending, setSending] = useState<boolean>(false)

  const { styles, width } = useStyle()

  useEffect(() => {
    configuration.recipients.forEach((recipient) => {
      const recipientName: string = recipient.name;
      setRecipients((previousRecipients) => {
        const updatedRecipients = {...previousRecipients, [recipientName]: recipient}

        if (recipientName === configuration.defaultRecipient) {
          recipient.selected = true;
          updatedRecipients[recipientName] = recipient
        }

        return updatedRecipients
      })
      
    });
  }, [])

  const toggleRecipientSelection = (recipientName: string) => {
    const recipient = recipients[recipientName];
    recipient.selected = !recipient.selected;
    setRecipients({...recipients, [recipientName]: recipient})
  }

  const showToast = (message: string, duration=ToastAndroid.SHORT) => {
    if (Platform.OS === 'android') {
      try {
        ToastAndroid.show(
          message,
          duration,
        );
        return
      } catch (error) {
        console.log(`error when showing toast - message: ${message}; error: ${error}`);
      }
    } else {
      console.log(`Not supported on platform: ${Platform.OS}`)
    }
  }

  const resetState = () => {
    setSubject('')
    setBody('')
    setSending(false)
  }

  const sendEmail = async () => {
    if (Object.keys(recipients).filter((name) => recipients[name].selected).length === 0) {
      showToast("At least one recipient needs to be selected.")
    }

    if (subject.trim().length === 0) {
      showToast("Subject is required.");
    }
    
    setSending(true)
    const recipientsArray = Object.keys(recipients).map((name) => recipients[name]);
    const addresses = recipientsArray.filter((recipient) => recipient.selected).map((recipient) => recipient.email);
    const errors: any[] = [];
    for (const address of addresses) {
      showToast("Sending email(s)...")
      try {
        await send(
          configuration.sender,
          address,
          subject,
          body);
      } catch (error) {
        errors.push(error);
      }
    }

    if (errors.length > 0) {
      console.log(`errors when sending email: ${JSON.stringify(errors, null, 2)}`);
      showToast(JSON.stringify(errors, null, 2), ToastAndroid.LONG)
    } else {
      showToast("Successfully sent email(s)")
    }
    resetState();
  }

  const emailButtons = Object.keys(recipients).map((name, index) => {
    const recipient = recipients[name];
    return <AwesomeButton
      key={recipient.id}
      onPress={() => toggleRecipientSelection(name)}
      accessibilityLabel={name}
      backgroundColor={recipient.selected ? blue.base01 : bases.base01}
      backgroundActive={recipient.selected ? blue.base02 : bases.base02}
      backgroundDarker={recipient.selected ? blue.base03 : bases.base03}
      disabled={sending}
    >
      {` ${name} `}
    </AwesomeButton>;
  });

  return (
    <ScrollView style={styles.root} 
      keyboardShouldPersistTaps='handled' >
        <View style={styles.emailForm} >
          <View style={styles.recipientButtonGroup}>{emailButtons}</View>
          <View>
            <TextInput
              // Subject Field
              onChangeText={(subject) => setSubject(subject)}
              style={styles.subject}
              value={subject}
              autoFocus={true}
              autoCapitalize="none"
              autoCorrect={false}
              testID="cd7b46a4-2d81-47bf-abc1-7142aba8a7b0"
            />
          </View>
          <View>
            <TextInput
              // Body Field
              multiline={true}
              numberOfLines={8}
              onChangeText={(body) => setBody(body)}
              style={styles.body}
              value={body}
              autoCapitalize="none"
              autoCorrect={false}
              testID="6b50a3ac-a102-4150-823c-e20d44f0c84d"
            />
          </View>
          <View style={styles.actionButtonGroup}>
            <AwesomeButton
              // Clear button
              onPress={() => resetState()}
              accessibilityLabel="Clear"
              width={width / 3}
              backgroundColor={sending ? bases.base01 : red.base01}
              backgroundActive={sending ? bases.base02 : red.base02}
              backgroundDarker={sending ? bases.base03 : red.base03}
              disabled={sending}
            >
              Clear
            </AwesomeButton>
            <AwesomeButton
              // Submit Button
              onPress={() => sendEmail()}
              accessibilityLabel="Send"
              testID="2d8395f6-03a5-4c61-9c3b-595143aec8bf"
              width={width / 3}
              backgroundColor={sending ? bases.base01 : green.base01}
              backgroundActive={sending ? bases.base02 : green.base02}
              backgroundDarker={sending ? bases.base03 : green.base03}
              disabled={sending}
            >
              Send
            </AwesomeButton>
          </View>
        </View>
      </ScrollView >
  )
}

const useStyle = () => {
  const dimensions = useWindowDimensions();
  const width = dimensions.width * .8;
  const marginTop = dimensions.height * .03;
  const marginBottom = dimensions.height * .03;

  const styles = StyleSheet.create({
    actionButtonGroup: {
      flexDirection: "row",
      justifyContent: "space-between",
      marginTop,
      marginBottom,
      width,
    },
    body: {
      paddingHorizontal: dimensions.width * .01,
      paddingVertical: dimensions.width * .01,
      backgroundColor: bases.base02,
      marginTop,
      textAlignVertical: "top",
      color: bases.base0,
      width,
    },
    recipientButtonGroup: {
      flexDirection: "row",
      justifyContent: "space-around",
      width
    },
    emailForm: {
      alignItems: "center",
      flex: 1,
      justifyContent: "flex-start",
      paddingTop: dimensions.height * .1,
    },
    root: {
      flex: 1,
    },
    subject: {
      paddingHorizontal: dimensions.width * .01,
      paddingVertical: dimensions.width * .01,
      backgroundColor: bases.base02,
      color: bases.base0,
      marginTop,
      width
    },
  });

  return { styles, width }
}
