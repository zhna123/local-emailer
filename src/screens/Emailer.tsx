import React, { useEffect, useState, useRef } from "react";
import {
  useWindowDimensions,
  StyleSheet,
  TextInput,
  ToastAndroid,
  View,
  ScrollView,
  Platform
} from "react-native";

import AwesomeButton from "react-native-really-awesome-button";

import { createBasesFromColor, rgb, rgbStrings as bases } from "solarizer";
import { configuration, Recipient } from "../Configuration";
import { sendEmail as send } from "../Google";
import AsyncStorage from '@react-native-async-storage/async-storage';
import { RootDrawerScreenProps } from "../../types";

const blue = createBasesFromColor(rgb.blue, "base01");
const red = createBasesFromColor(rgb.red, "base01");
const green = createBasesFromColor(rgb.green, "base01");

interface RecipientDict {
  [index: string]: Recipient
}

export const Emailer = ({ route, navigation }: RootDrawerScreenProps<'Emailer'>) => {
  
  const [recipients, setRecipients] = useState<RecipientDict>({})
  const [subject, setSubject] = useState<string>('')
  const [body, setBody] = useState<string>('')
  const [sending, setSending] = useState<boolean>(false)

  const subjectInput = useRef<TextInput>(null);

  const [defaultName, setDefaultName] = useState<string>('')

  const { styles, width } = useStyle()

  useEffect(() => {
    subjectInput.current?.focus()
  }, [subject])

  // get default recipient after initial render
  useEffect(() => {
    const getData = async () => {
      try {
        const value = await AsyncStorage.getItem('default-recipient');
        if (value !== null) {
          setDefaultName(value)
        }
      } catch (e) {
        console.log(`error reading async storage value: ${e}`)
      }
    };
    getData()
  }, [])

  // get all recipient after initial render
  useEffect(() => {
    configuration.recipients.forEach((recipient) => {
      const recipientName: string = recipient.name;
      setRecipients((previousRecipients) => {
        const updatedRecipients = {...previousRecipients, [recipientName]: recipient}
        return updatedRecipients
      })  
    });
  }, [])

  // update default recipient
  useEffect(() => {
    if (defaultName !== '') {
      const recipient = recipients[defaultName];
      recipient.selected = true;
      setRecipients({...recipients, [defaultName]: recipient})
    }
  }, [defaultName])

  const toggleRecipientSelection = (recipientName: string) => {
    const recipient = recipients[recipientName];
    recipient.selected = !recipient.selected;
    setRecipients({...recipients, [recipientName]: recipient})
  }

  const showToast = (message: string, duration=ToastAndroid.SHORT) => {
    if (Platform.OS === 'android') {
      try {
        ToastAndroid.show(
          message,
          duration,
        );
      } catch (error) {
        console.log(`error when showing toast - message: ${message}; error: ${error}`);
      }
    } else {
      console.log(`Not supported on platform: ${Platform.OS}`)
    }
  }

  const resetState = () => {
    setSubject('')
    setBody('')
    setSending(false)
  }

  const sendEmail = async () => {
    if (Object.keys(recipients).filter((name) => recipients[name].selected).length === 0) {
      showToast("At least one recipient needs to be selected.")
      return
    }

    if (subject.trim().length === 0) {
      showToast("Subject is required.");
      return 
    }
    
    setSending(true)
    const recipientsArray = Object.keys(recipients).map((name) => recipients[name]);
    const addresses = recipientsArray.filter((recipient) => recipient.selected).map((recipient) => recipient.email);
    const errors: any[] = [];
    for (const address of addresses) {
      showToast("Sending email(s)...")
      try {
        await send(
          configuration.sender,
          address,
          subject,
          body);
      } catch (error) {
        errors.push(error);
      }
    }

    if (errors.length > 0) {
      console.log(`errors when sending email: ${JSON.stringify(errors, null, 2)}`);
      showToast(JSON.stringify(errors, null, 2), ToastAndroid.LONG)
    } else {
      showToast("Successfully sent email(s)")
    }
    resetState();
  }

  const emailButtons = Object.keys(recipients).map((name, index) => {
    const recipient = recipients[name];

    return <AwesomeButton
      key={recipient.id}
      onPress={() => toggleRecipientSelection(name)}
      accessibilityLabel={name}
      backgroundColor={recipient.selected ? blue.base01 : bases.base01}
      backgroundActive={recipient.selected ? blue.base02 : bases.base02}
      backgroundDarker={recipient.selected ? blue.base03 : bases.base03}
      disabled={sending}
    >
      {` ${name} `}
    </AwesomeButton>;
  });

  return (
    <ScrollView style={styles.root} 
      contentContainerStyle={{flexGrow: 1}}
      keyboardShouldPersistTaps='handled' >
        <View style={styles.emailForm} >
          <View style={styles.recipientButtonGroup}>{emailButtons}</View>
          <View>
            <TextInput
              // Subject Field
              onChangeText={(subject) => setSubject(subject)}
              style={styles.subject}
              value={subject}
              ref={subjectInput}
              autoFocus={true}
              autoCapitalize="none"
              autoCorrect={false}
              blurOnSubmit={false}
              testID="cd7b46a4-2d81-47bf-abc1-7142aba8a7b0"
            />
          </View>
          <View>
            <TextInput
              // Body Field
              multiline={true}
              numberOfLines={8}
              onChangeText={(body) => setBody(body)}
              style={styles.body}
              value={body}
              autoCapitalize="none"
              autoCorrect={false}
              blurOnSubmit={true}
              testID="6b50a3ac-a102-4150-823c-e20d44f0c84d"
            />
          </View>
          <View style={styles.actionButtonGroup}>
            <AwesomeButton
              // Clear button
              onPress={() => resetState()}
              accessibilityLabel="Clear"
              width={width / 3}
              backgroundColor={sending ? bases.base01 : red.base01}
              backgroundActive={sending ? bases.base02 : red.base02}
              backgroundDarker={sending ? bases.base03 : red.base03}
              disabled={sending}
            >
              Clear
            </AwesomeButton>
            <AwesomeButton
              // Submit Button
              onPress={() => sendEmail()}
              accessibilityLabel="Send"
              testID="2d8395f6-03a5-4c61-9c3b-595143aec8bf"
              width={width / 3}
              backgroundColor={sending ? bases.base01 : green.base01}
              backgroundActive={sending ? bases.base02 : green.base02}
              backgroundDarker={sending ? bases.base03 : green.base03}
              disabled={sending}
            >
              Send
            </AwesomeButton>
          </View>
        </View>
      </ScrollView >
  )
}

const useStyle = () => {
  const dimensions = useWindowDimensions();
  const width = dimensions.width * .8;
  const marginTop = dimensions.height * .03;
  const marginBottom = dimensions.height * .03;

  const styles = StyleSheet.create({
    actionButtonGroup: {
      flexDirection: "row",
      justifyContent: "space-between",
      marginTop,
      marginBottom,
    },
    body: {
      paddingHorizontal: dimensions.width * .01,
      paddingVertical: dimensions.width * .01,
      backgroundColor: bases.base02,
      marginTop,
      textAlignVertical: "top",
      color: bases.base0,
    },
    recipientButtonGroup: {
      flexDirection: "row",
      justifyContent: "space-around",
    },
    emailForm: {
      flex: 1,
      paddingTop: 40,
      paddingHorizontal: 30
    },
    root: {
      backgroundColor: bases.base03,
    },
    subject: {
      paddingHorizontal: dimensions.width * .01,
      paddingVertical: dimensions.width * .01,
      backgroundColor: bases.base02,
      color: bases.base0,
      marginTop,
    },
  });

  return { styles, width }
}
