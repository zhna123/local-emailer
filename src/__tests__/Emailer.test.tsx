import React from "react";
import { fireEvent, render, screen, waitFor } from '@testing-library/react-native'
import renderer from "react-test-renderer";
import { recipientConfiguration } from "../Configuration/Recipients";
import { Emailer } from "../screens/Emailer";
import * as google from "../Google";
import AsyncStorage from "@react-native-async-storage/async-storage";

jest.mock("../Google", () => {
  const originalModule = jest.requireActual('../Google')
  return {
    __esModule: true,
    ...originalModule,
    sendEmail: jest.fn(() => console.log(`Mocked 'sendEmail' function called.`))
  }
})

jest.mock('react-native-really-awesome-button', () => "AwesomeButton");

describe("Emailer", () => {
  beforeEach(async () => {
    (google.sendEmail as any).mockClear();
    await waitFor(() => render(<Emailer />));
  });

  it("renders correctly", async () => {
    const tree = await waitFor(() => renderer.create(<Emailer />));
    const treeJson = tree.toJSON();
    expect(treeJson).toMatchSnapshot()
  });

  it('checks if Async Storage is used', async () => {
    await waitFor(() => 
      expect(AsyncStorage.getItem).toBeCalledWith('default-recipient')
    )
  });

  it("Should gather and send all fields", async () => {
    const testSubject = "Test Subject";
    const testBody = "Test Body";

    const subject = screen.getByTestId("cd7b46a4-2d81-47bf-abc1-7142aba8a7b0");
    fireEvent.changeText(subject, testSubject);

    const body = screen.getByTestId("6b50a3ac-a102-4150-823c-e20d44f0c84d");
    fireEvent.changeText(body, testBody);

    const sendButton = screen.getByTestId("2d8395f6-03a5-4c61-9c3b-595143aec8bf");
    // Why getByText doesn't work?
    // const sendButton = screen.getByText('Send')
    fireEvent.press(sendButton);

    const defaultRecipient = recipientConfiguration.recipients
      .find((recipient) => recipient.name === 'Mac');
    
    await waitFor(() => {
      expect(google.sendEmail).toHaveBeenCalledWith(
        recipientConfiguration.sender,
        defaultRecipient!.email,
        testSubject,
        testBody,
      );
    })
  });

  it("Should be able to handle chinese text", async () => {
    const testSubject = "我是一只猫";
    const testBody = "我有一个弟弟";

    const subject = screen.getByTestId("cd7b46a4-2d81-47bf-abc1-7142aba8a7b0");
    fireEvent.changeText(subject, testSubject);

    const body = screen.getByTestId("6b50a3ac-a102-4150-823c-e20d44f0c84d");
    fireEvent.changeText(body, testBody);

    const sendButton = screen.getByTestId("2d8395f6-03a5-4c61-9c3b-595143aec8bf");
    fireEvent.press(sendButton);

    const defaultRecipient = recipientConfiguration.recipients
      .find((recipient) => recipient.name === recipientConfiguration.defaultRecipient);

    await waitFor(() => {
      expect(google.sendEmail).toHaveBeenCalledWith(
        recipientConfiguration.sender,
        defaultRecipient!.email,
        testSubject,
        testBody,
      );
    })
  });
});