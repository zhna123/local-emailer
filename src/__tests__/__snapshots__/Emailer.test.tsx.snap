// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emailer renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "flexGrow": 1,
    }
  }
  keyboardShouldPersistTaps="handled"
  style={
    {
      "backgroundColor": "rgb(0,43,54)",
    }
  }
>
  <View>
    <View
      style={
        {
          "flex": 1,
          "paddingHorizontal": 30,
          "paddingTop": 133.4,
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-around",
          }
        }
      >
        <AwesomeButton
          accessibilityLabel="Mac"
          backgroundActive="rgb(-43,83,159)"
          backgroundColor="rgb(38,139,210)"
          backgroundDarker="rgb(-50,72,147)"
          disabled={false}
          onPress={[Function]}
        >
           Mac 
        </AwesomeButton>
        <AwesomeButton
          accessibilityLabel="Na"
          backgroundActive="rgb(7,54,66)"
          backgroundColor="rgb(88,110,117)"
          backgroundDarker="rgb(0,43,54)"
          disabled={false}
          onPress={[Function]}
        >
           Na 
        </AwesomeButton>
        <AwesomeButton
          accessibilityLabel="Eli"
          backgroundActive="rgb(7,54,66)"
          backgroundColor="rgb(88,110,117)"
          backgroundDarker="rgb(0,43,54)"
          disabled={false}
          onPress={[Function]}
        >
           Eli 
        </AwesomeButton>
      </View>
      <View>
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          autoFocus={true}
          onChangeText={[Function]}
          style={
            {
              "backgroundColor": "rgb(7,54,66)",
              "color": "rgb(131,148,150)",
              "marginTop": 40.019999999999996,
              "paddingHorizontal": 7.5,
              "paddingVertical": 7.5,
            }
          }
          testID="cd7b46a4-2d81-47bf-abc1-7142aba8a7b0"
          value=""
        />
      </View>
      <View>
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          multiline={true}
          numberOfLines={8}
          onChangeText={[Function]}
          style={
            {
              "backgroundColor": "rgb(7,54,66)",
              "color": "rgb(131,148,150)",
              "marginTop": 40.019999999999996,
              "paddingHorizontal": 7.5,
              "paddingVertical": 7.5,
              "textAlignVertical": "top",
            }
          }
          testID="6b50a3ac-a102-4150-823c-e20d44f0c84d"
          value=""
        />
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 40.019999999999996,
            "marginTop": 40.019999999999996,
          }
        }
      >
        <AwesomeButton
          accessibilityLabel="Clear"
          backgroundActive="rgb(139,-6,-4)"
          backgroundColor="rgb(220,50,47)"
          backgroundDarker="rgb(132,-17,-16)"
          disabled={false}
          onPress={[Function]}
          width={200}
        >
          Clear
        </AwesomeButton>
        <AwesomeButton
          accessibilityLabel="Send"
          backgroundActive="rgb(52,97,-51)"
          backgroundColor="rgb(133,153,0)"
          backgroundDarker="rgb(45,86,-63)"
          disabled={false}
          onPress={[Function]}
          testID="2d8395f6-03a5-4c61-9c3b-595143aec8bf"
          width={200}
        >
          Send
        </AwesomeButton>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
